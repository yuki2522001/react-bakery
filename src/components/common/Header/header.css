.header {
  display: flex;
  justify-content: space-between;
  padding: 80px 100px 0;
}

.header__about {
  display: flex;
}

.header__img {
  width: 100%;
  height: 500px;
}