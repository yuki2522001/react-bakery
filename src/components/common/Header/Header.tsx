import React from "react";
import MENU_LIST from "@constants/menu";
import url from "@assets/images/logos/logos.png";
import backgroundUrl from "@assets/images/backgrounds/slide-1.jpg";
import Logo from "../Logo/Logo";
import Menu from "../Menu/Menu";
import Button from "../Button/Button/Button";
import "./header.css";

const Header: React.FC = () => {
  return (
    <>
      <header data-testid="header" className="header">
        <div className="header__logo">
          <Logo src={url} />
        </div>
        <div className="header__about">
          <Menu menuList={MENU_LIST} />
          <Button type="primary" text="ORDER ONLINE" />
        </div>
      </header>
      <figure>
        <img src={backgroundUrl} className="header__img" alt="" />
      </figure>
    </>
  );
};

export default Header;
