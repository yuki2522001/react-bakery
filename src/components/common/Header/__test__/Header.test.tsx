import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import { createMemoryHistory } from "history";
import { Router } from "react-router-dom";
import Header from "../Header";

describe("Button component", () => {
  test("should render header component", () => {
    const history = createMemoryHistory();
    const { getByTestId } = render(
      <Router location={history.location} navigator={history}>
        <Header />
      </Router>
    );
    expect(getByTestId("header")).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const history = createMemoryHistory();
    const { asFragment } = render(
      <Router location={history.location} navigator={history}>
        <Header />
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});