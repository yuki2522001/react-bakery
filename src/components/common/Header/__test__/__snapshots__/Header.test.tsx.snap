// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="header"
    data-testid="header"
  >
    <div
      class="header__logo"
    >
      <a
        data-testid="logo"
        href="/"
      >
        <img
          class="logo"
        />
      </a>
    </div>
    <div
      class="header__about"
    >
      <div
        class="menu"
        data-testid="menu"
      >
        <ul
          class="menu__list"
        >
          <li
            class="menu__item"
          >
            <a
              href="/"
            >
              Home
            </a>
          </li>
          <li
            class="menu__item"
          >
            <a
              href=""
            >
              About
            </a>
          </li>
          <li
            class="menu__item"
          >
            <a
              href=""
            >
              Shop
            </a>
          </li>
          <li
            class="menu__item"
          >
            <a
              href=""
            >
              Blog
            </a>
          </li>
          <li
            class="menu__item"
          >
            <a
              href=""
            >
              FAQ
            </a>
          </li>
          <li
            class="menu__item"
          >
            <a
              href=""
            >
              Contact
            </a>
          </li>
        </ul>
      </div>
      <button
        class="btn btn__primary"
      >
        ORDER ONLINE
      </button>
    </div>
  </header>
  <figure>
    <img
      alt=""
      class="header__img"
    />
  </figure>
</DocumentFragment>
`;
